### PR TITLE
Deprecate DataApiMixin.

### PR DESCRIPTION
**Pull request by arkflpc**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Other (core): Deprecated `DataApiMixin` and moved `setData` and `getData` directly to the `Editor` class.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Comment by arkflpc** [comment]
I'm considering if it should be marked as breaking change. What is the probability someone used `DataApiMixin` on anything but `Editor`?
@niegowski ?

**Comment by filipsobol** [comment]
There's still one link to the `getData()` coming from `DataApiMixin` in the file `packages/ckeditor5-editor-decoupled/docs/framework/document-editor.md`. I think if was left by mistake.

**Review by filipsobol — COMMENTED** [review]
There's still one link to the `getData()` coming from `DataApiMixin` in the file `packages/ckeditor5-editor-decoupled/docs/framework/document-editor.md`. I think if was left by mistake. Everything else seems okay.

File: packages/ckeditor5-core/tests/editor/utils/dataapimixin.js [hunk 1]
```diff
@@ -5,73 +5,13 @@
...
-	} );
-
-	afterEach( async () => {
-		await editor.destroy();
-	} );
-
-	it( 'is compatible with `mix` function', () => {
-		expect( DataApiMixin ).have.property( 'setData' ).to.be.a( 'function' );
-		expect( DataApiMixin ).have.property( 'getData' ).to.be.a( 'function' );
-	} );
```
Comment (niegowski) [inline packages/ckeditor5-core/tests/editor/utils/dataapimixin.js:32]: I'd keep such a test to ensure we are backward compatible with the old `mix()` function.

**Review by niegowski — APPROVED** [review]

**Pull request merged**